# Lima & Callao - COVID Deaths Analysis

In [1]:
import numpy as np
import pandas as pd
from tqdm.auto import tqdm

In [2]:
df = pd.read_csv('Lima & Callo Covid Deaths Cleaned Dataset.csv')

In [3]:
df.head()

CUT OFF DATE  DEATH DATE  AGE     SEX    CLASSIFICATION DEPARTMENT PROVINCE  \
0   23-10-2023  15-04-2021   56  FEMALE  SINADEF Criteria       LIMA     LIMA   
1   23-10-2023  09-03-2021   73    MALE  SINADEF Criteria       LIMA     LIMA   
2   23-10-2023  19-08-2021   66    MALE  SINADEF Criteria       LIMA     LIMA   
3   23-10-2023  02-04-2021   57  FEMALE  SINADEF Criteria       LIMA     LIMA   
4   23-10-2023  16-04-2021   82  FEMALE  SINADEF Criteria       LIMA     LIMA   

               DISTRICT  UBIGEO        UUID  
0            CHORRILLOS  150108  13890096.0  
1         PUENTE PIEDRA  150125  13891240.0  
2            LURIGANCHO  150118  13891356.0  
3            LOS OLIVOS  150117  13891364.0  
4  SAN MARTIN DE PORRES  150135  13892353.0

In [4]:
# Convert 'Date_cort' and 'Death date' to datetime
# for column in tqdm(['Date_cort', 'Death date'], desc='Converting columns to datetime'):
#     df[column] = pd.to_datetime(df[column], format='%Y%m%d')

In [5]:
# df.head()

In [6]:
# Since the dates have been converted, let's save the dataframe with the updated date fields.
# df.to_csv('Updated_COVID_Deaths_Data.csv', index=False)
# print('The dataframe with updated date fields has been saved as Updated_COVID_Deaths_Data.csv.')

In [11]:
# Load the dataset
file_path = 'Lima & Callo Covid Deaths Cleaned Dataset.csv'
data = pd.read_csv(file_path, encoding='utf-8')

# Display the head of the dataframe
print(data.head())

# Generate distributions
for column in tqdm(['AGE', 'SEX', 'CLASSIFICATION', 'DEPARTMENT', 'PROVINCE', 'DISTRICT'], desc='Generating distributions'):
    print(f'Distribution for {column}:')
    print('---------------------------')
    print(data[column].value_counts(normalize=True).head())
    print('\n')

  CUT OFF DATE  DEATH DATE  AGE     SEX    CLASSIFICATION DEPARTMENT PROVINCE  \
0   23-10-2023  15-04-2021   56  FEMALE  SINADEF Criteria       LIMA     LIMA   
1   23-10-2023  09-03-2021   73    MALE  SINADEF Criteria       LIMA     LIMA   
2   23-10-2023  19-08-2021   66    MALE  SINADEF Criteria       LIMA     LIMA   
3   23-10-2023  02-04-2021   57  FEMALE  SINADEF Criteria       LIMA     LIMA   
4   23-10-2023  16-04-2021   82  FEMALE  SINADEF Criteria       LIMA     LIMA   

               DISTRICT  UBIGEO        UUID  
0            CHORRILLOS  150108  13890096.0  
1         PUENTE PIEDRA  150125  13891240.0  
2            LURIGANCHO  150118  13891356.0  
3            LOS OLIVOS  150117  13891364.0  
4  SAN MARTIN DE PORRES  150135  13892353.0  


Generating distributions:   0%|          | 0/6 [00:00<?, ?it/s]

Distribution for AGE:
---------------------------
AGE
67    0.026933
66    0.026656
72    0.026448
68    0.026220
70    0.025973
Name: proportion, dtype: float64


Distribution for SEX:
---------------------------
SEX
MALE      0.633303
FEMALE    0.366697
Name: proportion, dtype: float64


Distribution for CLASSIFICATION:
---------------------------
CLASSIFICATION
Virological Criteria                 0.418909
SINADEF Criteria                     0.328708
Serological Criteria                 0.169832
Epidemiological Research Criteria    0.029318
Clinical Criteria                    0.026111
Name: proportion, dtype: float64


Distribution for DEPARTMENT:
---------------------------
DEPARTMENT
LIMA      0.891299
CALLAO    0.108701
Name: proportion, dtype: float64


Distribution for PROVINCE:
---------------------------
PROVINCE
LIMA      0.891299
CALLAO    0.108701
Name: proportion, dtype: float64


Distribution for DISTRICT:
---------------------------
DISTRICT
LIMA                      

The dataset contains information about COVID-19 related deaths in Lima and Callao, including the date of death, age, sex, classification criteria, and location details.

The age distribution shows that the most common age at death is 67, representing approximately 2.69% of the dataset.

Males account for about 63.33% of the deaths, indicating a higher mortality rate among males compared to females.

The classification of deaths shows that 'Virological Criteria' is a common classification, accounting for approximately 41.89%.

The department distribution is heavily skewed towards Lima, with about 89.13% of the deaths occurring there.

Similarly, the province distribution mirrors the department distribution, with Lima also accounting for about 89.13% of the deaths.

The district distribution is more varied, with the Lima district itself accounting for approximately 8.81% of the deaths.